oxidation/libparsec/crates/platform-async/src/native/timer.rs

```diff
@@ -0,0 +1,44 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use futures::FutureExt;
+use tokio::time::{sleep, Sleep};
+
+pub struct Timer {
+    sleep: Pin<Box<Sleep>>,
+}
+
+impl Timer {
+    /// Create a timer that expires after the given duration of time.
+    ///
+    /// ```
+    /// use std::time::{Duration, Instant};
+    /// use libparsec_platform_async::Timer;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// let timer = Instant::now();
+    /// Timer::after(Duration::from_millis(100)).await;
+    /// assert!(timer.elapsed() < Duration::from_millis(110));
+    /// # }
+    /// ```
+    pub fn after(duration: Duration) -> Timer {
+        Self {
+            sleep: Box::pin(sleep(duration)),
+        }
+    }
+}
+
+impl Future for Timer {
+    type Output = ();
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        self.sleep.poll_unpin(cx)
```

Comment:
Use `Pin::new(self).poll(cx)` to avoid needed future ? (cf. https://rust-lang-nursery.github.io/futures-api-docs/0.1.27/src/futures_util/future/mod.rs.html#531)

Reply:
I prefer ctx instead of cx, because it is used in a lot of crates

Reply:
`cx` is the default name of the parameters in the trait definition

Reply:
https://doc.rust-lang.org/std/future/trait.Future.html

Reply:
God :man_facepalming: